app/controllers/simplified-account/settings/card-types/card-types.controller.js

```diff
@@ -1,22 +1,74 @@
 const { response } = require('@utils/response')
 const { formatCardTypesForTemplate } = require('@utils/simplified-account/format/format-card-types')
-const { getAllCardTypes, getAcceptedCardTypesForServiceAndAccountType } = require('@services/card-types.service')
+const { getAllCardTypes, getAcceptedCardTypesForServiceAndAccountType, postAcceptedCardsForServiceAndAccountType } = require('@services/card-types.service')
+const { formatSimplifiedAccountPathsFor } = require('@utils/simplified-account/format')
+const paths = require('@root/paths')
 
 async function get (req, res, next) {
   const serviceId = req.service.externalId
   const accountType = req.account.type
   const isAdminUser = req.user.isAdminUserForService(serviceId)
+  const messages = res.locals?.flash?.messages ?? []
+  let noCardTypesSelectedError
+  if (res.locals?.flash?.noCardTypesSelectedError) {
+    noCardTypesSelectedError = { summary: [{ text: res.locals?.flash?.noCardTypesSelectedError }] }
+  }
   try {
     const { card_types: allCards } = await getAllCardTypes()
     const { card_types: acceptedCards } = await getAcceptedCardTypesForServiceAndAccountType(serviceId, accountType)
-    const cardTypes = formatCardTypesForTemplate(allCards, acceptedCards, req.account, isAdminUser)
-    response(req, res, 'simplified-account/settings/card-types/index',
-      { cardTypes, isAdminUser })
+    const cardTypesSelected = noCardTypesSelectedError ? [] : acceptedCards
+    const currentAcceptedCardTypeIds = acceptedCards.map(card => card.id)
+    const cardTypes = formatCardTypesForTemplate(allCards, cardTypesSelected, req.account, isAdminUser)
+    response(req, res, 'simplified-account/settings/card-types/index', {
+      errors: noCardTypesSelectedError,
+      messages,
+      cardTypes,
+      isAdminUser,
+      currentAcceptedCardTypeIds
+    })
+  } catch (err) {
+    next(err)
+  }
+}
+
+async function post (req, res, next) {
+  console.log(req.body)
+  const serviceId = req.service.externalId
+  const accountType = req.account.type
+  const selectedDebitCards = (typeof req.body.debit === 'string' ? [req.body.debit] : req.body.debit) || []
+  const selectedCreditCards = (typeof req.body.credit === 'string' ? [req.body.credit] : req.body.credit) || []
+  const selectedCardTypeIds = [...selectedDebitCards, ...selectedCreditCards]
+  const currentAcceptedCardTypeIds = req.body.currentAcceptedCardTypeIds ? req.body.currentAcceptedCardTypeIds.split(',') : []
+  if (!selectedDebitCards.length && !selectedCreditCards.length) {
+    req.flash('noCardTypesSelectedError', 'You must choose at least one card')
```

Comment:
is there a reason you're using flash to return errors here instead the express validate chain / error response approach we use everywhere else?

Reply:
seemed more DRY than having to reload all the accepted card types again

Reply:
updated to use the same pattern as suggested.